File: backend/DTOs/TransactionAdminDto.cs
using System;

namespace backend.DTOs
{
    public class TransactionAdminDto
    {
        public int Id { get; set; }

        public int UserId { get; set; }
        public string UserName { get; set; } = string.Empty;
        public long TgUserId { get; set; }

        /// <summary>Null when the transaction is a manual points adjustment.</summary>
        public int? TravelId { get; set; }
        public string TravelTitle { get; set; } = string.Empty;

        /// <summary>Signed points moved on the user's balance.</summary>
        public int Points { get; set; }

        public DateTimeOffset CreatedAt { get; set; }

        /// <summary>False when the transaction has been voided (soft deleted).</summary>
        public bool Status { get; set; }
    }
}

File: frontend/src/app/pages/desktop/transactions/models/transaction-admin-dto.ts
export interface TransactionAdminDto {
  readonly id: number;

  readonly userId: number;
  readonly userName: string;
  readonly tgUserId: number;

  /** Null when the transaction is a manual points adjustment. */
  readonly travelId: number | null;
  readonly travelTitle: string;

  /** Signed points moved on the user's balance. */
  readonly points: number;

  readonly createdAt: string;

  /** Soft-delete flag. `false` means the transaction has been voided. */
  readonly status: boolean;
}

/** Body accepted by the admin create and update endpoints. */
export interface TransactionPayload {
  readonly userId: number;

  /** Null for a manual points adjustment. */
  readonly travelId: number | null;

  /** Signed points: positive adds to the balance, negative subtracts. */
  readonly points: number;

  readonly status: boolean;
}
